### ISSUE TITLE
Confusing error message when trying to call rust-lldb on windows

### ISSUE THREAD

**Issue by skade** [rust-lang/rustup#2233]
**Problem**

This is a problem I caught on twitter: https://twitter.com/robdaemon/status/1229188704666820608

Calling `rust-lldb` on a windows toolchain msvc seems to result in the following interaction:

```
rust-lldb.exe is not installed...
To install, run `rustup component add lldb-preview`.
```
Running the command results in:

```
component 'lldb-preview' for target x86-64-pc-windows-msvc...
```

This interaction is not useful, especially as rust-lldb is not available for this target.

**Possible Solution(s)**

My suggested solution would be to error out early with a message that MSVC targets are not supported by lldb.

**Comment by robdaemon** [comment]
![rust-lldb](https://user-images.githubusercontent.com/287213/74615406-45f28800-50d5-11ea-9869-f608b4636a43.png)

adding a screenshot here, as my Twitter feed gets purged regularly.

**Comment by kinnison** [comment]
The useful change for this would be to ensure that when raising the `BinaryNotFound` from `Toolchain::create_command()` we should check the toolchain's available components and add a boolean to the raise when the command would not be available in the toolchain.  This would have the added bonus that in nightlies where `rustfmt` or `clippy` is unavailable, we'd explain that up-front rather than risking confusion later.

If someone wishes to tackle this, I'd be pleased to help them find the right places to make such a change.

**Comment by virome** [comment]
Hello kinnison, I would like to take this one, if possible.

**Comment by rbtcollins** [comment]
I think an enum would be more useful.
Result<ToolchainCommand>

enum ToolchainCommand {
UnknownBinary
BinaryNotBuilt
Command(command)
}

Where errors indicate failures like the toolchain isn't actually installed, etc, Command(command) holds the built command, and we can tell the difference between a binary that should exist but is temporarily missing, and permanently - if the manifest has that info?
o

**Comment by virome** [comment]
I was wondering along the same lines. Possibly a new error, maybe something like `CommandNotSupportedOnToolchain` that would be for instances when we've identified it as a command that's commonly available on other platforms but not the current one (how we would do that, I'm not sure yet), whereas `BinaryNotFound` would be for FS level errors.

**Comment by kinnison** [comment]
@virome Hi and I'm excited you'd like to try and fix this.  I'm glad you're agreeing with the approach that @rbtcollins is suggesting, and if you want further discussion on these topics then there is `#wg-rustup` on the rust-lang discord server https://discord.gg/rust-lang

**Comment by kinnison** [comment]
I believe this is closed in #2264 and thus will be in 1.22.0
